src/main/java/com/amazon/dlic/auth/http/jwt/keybyoidc/HTTPOpenIdAuthenticator.java

```diff
@@ -0,0 +1,328 @@
+            missing = missing.concat(SUB_CLAIM);
+        }
+
+        if (isSigned) {
+            if (claims.getIssuer() == null || claims.getIssuer().isBlank() || !claims.getIssuer().equals(settings.get(ISSUER_ID_URL))) {
+                missing = missing.concat("iss");
+            }
+            if (claims.getAudience() == null
+                || claims.getAudience().toString().isBlank()
+                || !claims.getAudience().contains(settings.get(CLIENT_ID))) {
+                missing = missing.concat("aud");
+            }
+        }
+
+        return missing;
+    }
+
+    private final class HTTPJwtKeyByOpenIdConnectAuthenticator extends AbstractHTTPJwtAuthenticator {
+
+        public HTTPJwtKeyByOpenIdConnectAuthenticator(Settings settings, Path configPath) {
+            super(settings, configPath);
+        }
+
+        protected KeyProvider initKeyProvider(Settings settings, Path configPath) throws Exception {
+            int idpRequestTimeoutMs = settings.getAsInt("idp_request_timeout_ms", 5000);
+            int idpQueuedThreadTimeoutMs = settings.getAsInt("idp_queued_thread_timeout_ms", 2500);
+
+            int refreshRateLimitTimeWindowMs = settings.getAsInt("refresh_rate_limit_time_window_ms", 10000);
+            int refreshRateLimitCount = settings.getAsInt("refresh_rate_limit_count", 10);
+            String jwksUri = settings.get("jwks_uri");
+            KeySetRetriever keySetRetriever;
+
+            if (jwksUri != null && !jwksUri.isBlank()) {
+                keySetRetriever = new KeySetRetriever(
+                    getSSLConfig(settings, configPath),
+                    settings.getAsBoolean("cache_jwks_endpoint", false),
+                    jwksUri
+                );
+            } else {
+                keySetRetriever = new KeySetRetriever(
+                    settings.get("openid_connect_url"),
+                    getSSLConfig(settings, configPath),
+                    settings.getAsBoolean("cache_jwks_endpoint", false)
+                );
+            }
+
+            keySetRetriever.setRequestTimeoutMs(idpRequestTimeoutMs);
+
+            SelfRefreshingKeySet selfRefreshingKeySet = new SelfRefreshingKeySet(keySetRetriever);
+
+            selfRefreshingKeySet.setRequestTimeoutMs(idpRequestTimeoutMs);
+            selfRefreshingKeySet.setQueuedThreadTimeoutMs(idpQueuedThreadTimeoutMs);
+            selfRefreshingKeySet.setRefreshRateLimitTimeWindowMs(refreshRateLimitTimeWindowMs);
+            selfRefreshingKeySet.setRefreshRateLimitCount(refreshRateLimitCount);
+
+            return selfRefreshingKeySet;
+        }
+
+        private JWTClaimsSet getJwtClaimsSet(SecurityRequest request) throws OpenSearchSecurityException {
+            String parsedToken = super.getJwtTokenString(request);
+            return getJwtClaimsSetFromInfoContent(parsedToken);
+        }
+
+        private JWTClaimsSet getJwtClaimsSetFromInfoContent(String userInfoContent) throws OpenSearchSecurityException {
```

Comment:
This is new--how I split it out from the extract method--but the rest of the nested class was previously implemented.